CRM/Case/BAO/Case.php

```diff
@@ -529,8 +529,14 @@ public static function retrieveCaseIdsByContactId($contactID, $includeDeleted =
     if (!$includeDeleted) {
       $query .= " AND ca.is_deleted = 0";
     }
+    if (isset($statusId)) {
+      $params = array(1 => array($contactID, 'Integer'), 2 => array($statusId, 'Integer'));
+      $query .= " AND ca.status_id = %2";
+    }
+    else {
+      $params = array(1 => array($contactID, 'Integer'));
+    }
```

Comment:
This will do the trick, but is a bit repetitive and this approach would quickly get very very messy with > 2 params.
My suggestion was to move the line ` $params = array(1 => array($contactID, 'Integer'));` to the top of the function, and inside the above conditional put `$params[2] = array($statusId, 'Integer');`